src/main/java/com/mycompany/shop/basket/BasketController.java

```diff
@@ -33,21 +39,75 @@
     static class CreateBasketRequest {
         Long id;
 
+        Long userId;
+
         public Long getId() {
             return id;
         }
 
         public void setId(Long id) {
             this.id = id;
         }
+
+        public Long getUserId() {
+            return userId;
+        }
+
+        public void setUserId(Long userId) {
+            this.userId = userId;
+        }
     }
 
     @GetMapping
+    public ResponseEntity<Basket> getUserBasket(@RequestParam("userId") Long userId) {
+        Basket basket;
+        if (userId != null) {
+            if (basketRepository.findBasketByUserId(userId).isPresent()) {
+                basket = basketRepository.findBasketByUserId(userId).get();
+                sessionBean.setBasket(basket);
+            } else {
+                sessionBean.setBasket(null);
+            }
+        }
+        return ResponseEntity.ok(sessionBean.getBasket());
+    }
+
+    @GetMapping("/unregistered")
     public ResponseEntity<Basket> getBasket() {
         return ResponseEntity.ok(sessionBean.getBasket());
     }
 
     @PostMapping
+    public ResponseEntity<Basket> addToUserBasket(@RequestBody CreateBasketRequest createBasketRequest) {
```

Comment:
you don't need additional rest method for adding into basket for authorized user. use one method and build the logic inside the method based on information from session if user is logged in or not

Reply:
+